rest/src/main/kotlin/systems/ajax/englishstudytelegrambot/external/source/ExternalWordSource.kt

```diff
@@ -1,39 +1,25 @@
 package systems.ajax.englishstudytelegrambot.external.source
 
-import org.slf4j.LoggerFactory
-import org.springframework.http.HttpStatus
 import org.springframework.stereotype.Component
 import org.springframework.web.reactive.function.client.WebClient
-import org.springframework.web.server.ResponseStatusException
-import systems.ajax.englishstudytelegrambot.dto.GettingPartOfAdditionalInfoAboutWord
+import reactor.core.publisher.Mono
+import systems.ajax.englishstudytelegrambot.dto.external.api.GettingPartOfAdditionalInfoAboutWord
 import systems.ajax.englishstudytelegrambot.property.WordnikProperties
 
 @Component
 class ExternalWordSource(val wordnikProperties: WordnikProperties, val webClient: WebClient) {
 
-    final inline fun <reified T : GettingPartOfAdditionalInfoAboutWord> customGetInfoAboutWord(
+    final inline fun <reified T : GettingPartOfAdditionalInfoAboutWord> customGetInfoAboutWordFromWordnikAPI(
         wordSpelling: String,
         link: String
-    ): String {
-        return runCatching {
-            webClient.get()
-                .uri(
-                    link, wordSpelling, wordnikProperties.tokenKey
-                )
-                .retrieve()
-                .onStatus({ responseStatus ->
-                    responseStatus == HttpStatus.NOT_FOUND
-                }) { throw ResponseStatusException(HttpStatus.NOT_FOUND) }
-                .bodyToFlux(T::class.java)
-                .blockFirst()!!
-                .partOfAdditionalInfoAboutWord
-        }.getOrElse {
-            log.info("additional info for word {} isn't found", wordSpelling)
-            "Missing"
-        }
-    }
-
-    companion object {
-        val log = LoggerFactory.getLogger(this::class.java)
-    }
+    ): Mono<String> =
+        webClient.get()
+            .uri(
+                link, wordSpelling, wordnikProperties.tokenKey
+            )
+            .retrieve()
+            .bodyToFlux(T::class.java)
+            .next()
+            .map(GettingPartOfAdditionalInfoAboutWord::partOfAdditionalInfoAboutWord)
+            .onErrorReturn("Missing")
```

Comment:
```kotlin
.doOnError {
    log.error("additional info for word {} isn't found", wordSpelling)
}
```
?

Reply:
sometime wordnik doesnt return answer and in that case i put Missing like an answer

Reply:
but if you mean log when error happens, i agree and add